### ISSUE TITLE
bug: Input with inline label goes off-screen

### ISSUE THREAD

**Issue by mikemintz** [driftyco/ionic#2903]
**Type**: <span ionic-type>bug</span>

**Platform**: <span ionic-platform>all</span>  

<span ionic-description>When I have a form with inline labels, the label goes off-screen when I fill up the text field. Below are screenshots before and after typing text into my browser (300px wide).

I'm using markup specified at http://ionicframework.com/docs/components/#forms-inline-labels which looks like

    <div class="list">
      <label class="item item-input">
        <span class="input-label">Email address</span>
        <input type="email" placeholder="Enter email">
      </label>
    </div>

Reproduced here http://codepen.io/mikemintz/pen/LExZYb?editors=101 You have to make the browser window narrow enough to see this. I can reproduce it on an iPhone 5 by loading it directly at http://codepen.io/mikemintz/debug/LExZYb

Also reported on the forum: http://forum.ionicframework.com/t/input-with-inline-label-goes-off-screen/15271</span>

<span is-issue-template></span>

**Comment by pulakb** [comment]
<div class="list">
  <label class="item item-input">
    <span class="input-label">Email address</span>
  </label>
<input type="email" placeholder="Enter email">
</div>

Try above and also can follow the following tutorial:
http://www.sitepoint.com/position-text-labels-on-forms-using-css/

**Comment by mikemintz** [comment]
@pulakb if I put the input field outside the label element, the input no longer appears on the same line with the label text.

**Comment by pulakb** [comment]
@mikemintz Yes I agree but that you can manage using CSS. You can go through the link I have shared above in that direction.

**Comment by mhartington** [comment]
Looks like we may need to change some flex properties.

http://codepen.io/mhartington/pen/zxNMqG

**Comment by bkocik** [comment]
The flex properties fix seems to work wonderfully on smaller screens like the iPhone 4 and 5, but then breaks on the iPhone 6 and 6+.

**Comment by mhartington** [comment]
Hmm, the only solution I could come up with is changing the whole flex property to this.
```
.item-input input{
  flex: 1 0 65%;
  text-overflow: ellipsis;
}

.input-label {
  flex: 1 0 35%;
}
```
http://codepen.io/mhartington/pen/gbePEr

But the main reason this happens is that the length of the input and span is larger than the whole list item.

It may be worth looking into switching everything to percent values, but I don't think that will happen with 1.0 coming up soon. 

@adamdbradley something to note for post 1.0

@mikemintz if you're not up for writing tweaking some css on your end, you may want to consider stacked labels.

http://ionicframework.com/docs/components/#forms-stacked-labels

**Comment by perrygovier** [comment]
fixed in d68ecc16b729ad361a655c1050aa03fe22aef137
